### ISSUE TITLE
RT fails with segfault

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/MAESTROeX#319]
Building the `rt` problem in non-debug mode and running with `inputs_2d_regression` generates a segfault before any steps are done:

```
$  ./Maestro2d.gnu.OMP.ex inputs_2d_regression 
OMP initialized with 8 OMP threads
AMReX (22.07-8-ga633d2bff1db) initialized
Calling Setup()
Calling ReadParameters()
reading extern runtime parameters ...
Calling VariableSetup()
Calling BCSetup()
Calling BaseStateGeometry::Init()
Calling Init()
Calling InitData()
initdata model_File = 
cutoff densities:
    low density cutoff (for mapping the model) =      1e-10
    buoyancy cutoff density                           
        (for zeroing rho - rho_0, centrifugal term) = 5e-10
    anelastic cutoff =                                1e-10
 
 
Maximum HSE Error = 4.547473509e-13
   (after putting initial model into base state arrays, and
    for density < base_cutoff_density)
 
Segfault
Segmentation fault (core dumped)
```

**Comment by zingale** [comment]
running through valgrind shows:

```
==424395== Invalid read of size 8
==424395==    at 0x4CD5B7: operator() (MaestroInitData.cpp:80)
==424395==    by 0x4CD5B7: call_f<Maestro::InitLevelData(int, amrex::Real, const amrex::MFIter&, amrex::Array4<double>, amrex::Array4<double>)::<lambda(int, int, int)> > (AMReX_GpuLaunchFunctsC.H:29)
==424395==    by 0x4CD5B7: ParallelFor<Maestro::InitLevelData(int, amrex::Real, const amrex::MFIter&, amrex::Array4<double>, amrex::Array4<double>)::<lambda(int, int, int)> > (AMReX_GpuLaunchFunctsC.H:119)
==424395==    by 0x4CD5B7: Maestro::InitLevelData(int, double, amrex::MFIter const&, amrex::Array4<double>, amrex::Array4<double>) (MaestroInitData.cpp:68)
==424395==    by 0x4BDA95: Maestro::MakeNewLevelFromScratch(int, double, amrex::BoxArray const&, amrex::DistributionMapping const&) [clone ._omp_fn.0] (MaestroInit.cpp:379)
==424395==    by 0x4EB75CD: gomp_thread_start (team.c:129)
==424395==    by 0x4F90E2C: start_thread (pthread_create.c:442)
==424395==    by 0x50157D3: clone (clone.S:100)
==424395==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
```

**Comment by zingale** [comment]
this has been fixed in #349
